practice.rb

```diff
@@ -15,26 +28,42 @@ def pal(s)
 raise "fib broke - fib(2)" unless fib(2) == 1
 puts "passes all fibanocci tests"
 
-# Palindrome Tests
+# # Palindrome Tests
 raise "pal broke - pal('racecar')" unless pal("racecar") == true
 raise "pal broke - pal('helloworld')" unless pal("helloworld") == false
 raise "pal broke - pal('')" unless pal("") == true
 puts "passes all palindrome tests"
 
+def power(base,exp)
+  if exp==1
+    return base
+  else
+    return base * power(base, exp - 1)
+  end
+
+end
+
 
+def binary(n)
+  if n == 0
+    return 0
+  elsif n == 1
+    return 1
+  end
+  if n.odd?
+    return 1 + binary(n-1)
+  elsif  n % 2 ** x == 0
+    return binary(n-1) + (10 ** x - 1)
+  end
+end 
```

Comment:
This one is hard because it's actually a string return and I didn't cover binary with you all. I did with previous cohorts - but I will cover it later in the curriculum with you all.